Protocol.md

```diff
@@ -23,188 +44,182 @@ The following figure gives a brief overview of the protocol:
-Example:
-`public_key` - `R_point`<sub>1</sub> - `R_point`<sub>2</sub> - `meta data`
-The meta data can either be human readable for a user to be able to interpret it, or machine readable so that an application can interpret it and present it to the user.
-If both parties already have this information, transmission is unnecessary.
+Note that multiple contracts can be open in parallel, as all DLC
+messages are identified by either a `temporary_contract_id` (before the
+funding transaction is created) or a `contract_id` (derived from the
+funding transaction).
+
+### The `offer_dlc` Message
+
+This message contains information about a node and indicates its
+desire to open a new contract. This is the first step toward creating
+the funding transaction and CETs.
+
+1. type: ??? (`offer_dlc`)
+2. data:
+   * [`byte`:`contract_flags`]
+   * [`chain_hash`:`chain_hash`]
+   * [`contract_info`:`contract_info`]
+   * [`oracle_info`:`oracle_info`]
+   * [`point`:`funding_pubkey`]
+   * [`spk`:`payout_spk`]
+   * [`u64`:`total_collateral_satoshis`]
+   * [`u16`:`num_funding_inputs`]
+   * [`num_funding_inputs*funding_input`:`funding_inputs`]
+   * [`spk`:`change_spk`]
+   * [`u64`:`feerate_per_vb`]
+   * [`u32`:`contract_maturity_cltv`]
+   * [`u32`:`contract_timeout_cltv`]
 
-#### A's fund amount (satoshi)
-How much A inputs into the contract.
+No bits of `contract_flags` are currently defined, this field should be ignored.
 
-#### A's public keys
-The following public keys:
+The `chain_hash` value denotes the exact blockchain that the DLC will
+reside within. This is usually the genesis hash of the respective blockchain.
+The existence of the `chain_hash` allows nodes to open contracts
+across many distinct blockchains as well as have contracts within multiple
+blockchains opened to the same peer (if it supports the target chains).
 
-1. Funding Public Key
-   - A's public key used in the multi-signature [funding output](Transactions.md#FundingOutputs)
-2. Sweep Public Key
-   - A's public key used in the creation of the CETs [output script](Transactions.md#CETOutputs).
+`contract_info` specifies the sender's payouts for all events. `oracle_info`
+specifies the oracle(s) to be used as well as their commitments to events.
 
-#### A's addresses
+`funding_pubkey` is the public key in the 2-of-2 multisig script of
+the funding transaction output. `payout_spk` specifies the script
+pubkey that CETs and the refund transaction should use in the sender's output.
 
-The following addresses:
+`total_collateral_satoshis` is the amount the sender is putting into the
+contract. `num_funding_inputs` is the number of funding inputs contributed by
+the sender and `funding_inputs` contains outputs, outpoints, and expected weights
+of the sender's funding inputs. `change_spk` specifies the script pubkey that funding
+change should be sent to.
```

Comment:
I could be easier to define `input` as `txid+vout` and then give an expected weight over the set of inputs?

Reply:
Note to self, look up with that new dual funding does with this

Reply:
https://github.com/discreetlogcontracts/dlcspecs/issues/72